File: leetcode/268_missing_number/main.py
def missing_num(nums: list[int]) -> int:
    n = len(nums)

    sum = 0
    for i in range(n+1):
        sum += i

    for num in nums:
        sum -= num

    return sum     


print(missing_num([0, 1, 3, 4]))
print(missing_num([0, 4, 1, 3]))
print(missing_num([2, 4, 1, 3]))

File: leetcode/268_missing_number/main.go
package main

import "fmt"

func main() {
	fmt.Println(missingNumber([]int{1, 2, 3}))
	fmt.Println(missingNumber([]int{3, 2, 1}))
	fmt.Println(missingNumber([]int{3, 0, 1}))
}

func missingNumber(nums []int) int {
	n := len(nums)

	var sum int
	for i := 0; i < n+1; i++ {
		sum += i
	}	

	for _, v := range nums {
		sum -= v
	}

	return sum
}